test/files/run/repl-type.check

```diff
@@ -0,0 +1,11 @@
+
+scala> :type 42
+Int
+
+scala> val x: 23 = 23
+val x: 23 = 23
+
+scala> :type x
+23
+
+scala> :quit
```

Comment:
should this also test something like `:type (23: 23)`?

Reply:
No because it doesn't work :p 

```scala
scala> def fresh = (23: 23)
def fresh: Int

scala> :type (23: 23)
Int
```

Reply:
Added the test here - 8a1deee

Reply:
IMHO, this fix is very limited, it won't work also for sth like this:

```scala
val x: 23 = 23
val y = x
```

Reply:
@eed3si9n fair enough. *should* it also work though?

Reply:
@NthPortal Yes. Since `:type` is supposed to display the type information of any expressions, I definitely think it should work. It's a good example because it exemplifies the limitation of current "evaluate on fresh val or def" approach.